### PR TITLE
Result Table

### PR DESCRIPTION
**Pull request by denisgursky**
You can run the command below to see it in action. The results will be displayed using `table.print()`
```
npm run example -- ./examples/execAsync.ts -d DB_NAME -e ENGINE_NAME -c 'def output = 123' --poll
```

### CODE REVIEW

File: src/results/resultTable.ts [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+      return colDef;
+    });
+  }
+
+  /**
+   * Array for type definitions per column. Shortcut for column.typeDef.
+   *
+   * @returns An array of type definitions.
+   */
+  get typeDefs() {
```
Comment (larf311) [inline src/results/resultTable.ts:118]: Should this just be a function instead of a getter? Seems similar to `columns` to me.

**Comment by larf311** [comment]
Looks pretty awesome! I will say that the function naming is very JS oriented. For example, I don't think `slice` is a common name in other languages. [B2T2](https://github.com/brownplt/B2T2/blob/main/TableAPI.md) seems to use `select` and you pass in an array of indexes. Java's ArrayList uses `subList`. Python uses slice but on the array itself, not a function. I guess we are shooting for being idiomatic in each language so I suppose these names can diverge. 🤷

File: src/results/resultTable.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
+/**
+ * Copyright 2021 RelationalAI, Inc.
```
Comment (larf311) [inline src/results/resultTable.ts:2]: It's 2022!

**Comment by denisgursky** [comment]
> Looks pretty awesome! I will say that the function naming is very JS oriented. For example, I don't think `slice` is a common name in other languages. [B2T2](https://github.com/brownplt/B2T2/blob/main/TableAPI.md) seems to use `select` and you pass in an array of indexes. Java's ArrayList uses `subList`. Python uses slice but on the array itself, not a function. I guess we are shooting for being idiomatic in each language so I suppose these names can diverge. 🤷

[select](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html#select) is used for column selection in the Arrow SDK. I was thinking about doing the same here instead of `sliceColumns` maybe.

Arrow API seems to be pretty different from language to language https://github.com/apache/arrow. I think it's fine to diverge and be idiomatic as long as all SDKs implement the same set of functionality.

File: src/results/resultTable.ts (hunk 2 above)
Comment (denisgursky) [reply src/results/resultTable.ts:2]: Yeah, probably a separate PR with a mass change for all files.

**Review by larf311 — APPROVED** [review]

**Review by NRHelmi — APPROVED** [review]
Looks great to me thanks @denisgursky

**Comment by vilterp** [comment]
Excited for this. One naming bikeshed comment — since this will eventually be the primary result type, do we want to call it something high-level and Rel-y like `Relation`?

**Comment by larf311** [comment]
@vilterp I thought that too at first but according to wikipedia a relation implies no ordering. But our result set has ordering, so I think table is the proper term.

**Comment by denisgursky** [comment]
Also we still export type named as `Relation` which is v1 json relation object.

**Review by vilterp — APPROVED** [review]

**Pull request merged**